map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -515,17 +554,95 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
-                        map.animateCamera(CameraUpdateFactory.newLatLngZoom(marker.position, Math.max(ZOOM_LEVEL_THREE, currentZoomLevel)))
+                    fullObjectsLock.withLock {
+                        fullObjectMarkers.firstOrNull { (model, _) ->
+                            model is MapItem.Object && model.item.nid == nid
+                                    ||
+                                    model is MapItem.TourIntro && model.item.nid == nid
+                        }?.let { (_, marker) ->
+                            val currentZoomLevel = map.cameraPosition.zoom
+                            map.animateCamera(CameraUpdateFactory.newLatLngZoom(marker.position, Math.max(ZOOM_LEVEL_THREE, currentZoomLevel)))
+                        }
                     }
 
                 }.disposedBy(disposeBag)
     }
 
+    @UiThread
+    fun bindToMap(mapItem: MapItem<*>, mapMode: MapViewModel.DisplayMode) {
+        when (mapItem) {
+            is MapItem.Annotation -> {
+                val annotation = mapItem.item
+                when (annotation.annotationType) {
+                    ArticMapAnnotationType.DEPARTMENT -> {
+                        loadDepartment(mapItem)
+                    }
+                }
+            }
+            is MapItem.Gallery -> {
+                loadGalleryNumber(mapItem)
+            }
+            is MapItem.Object -> {
+                loadObject(mapItem, mapMode)
+            }
+            is MapItem.TourIntro -> {
+                loadTourObject(mapItem, mapMode)
+            }
+
+        }
+    }
+
+
+    @WorkerThread
+    fun bindMarkersAsynchronously(itemList: List<MapItem<*>>, mapMode: MapViewModel.DisplayMode) {
+        val coreCount = Runtime.getRuntime().availableProcessors()
+
+        val batchedItems : List<List<MapItem<*>>>
+        batchedItems = if (requireContext().isResourceConstrained()) {
+            // We don't have that many resources. Only pass along 2 x 'number of cores' items at a time
+            itemList.chunked(2 * coreCount)
+        } else {
+            // Work with 3 x 'number of processor cores' items at a time.
+            itemList.chunked(3 * coreCount)
+        }
+
+                Observables.zip(
+                        // There's an Observable operator 'window', but that's (severely) restricted to a single subscriber
+                        batchedItems.toObservable(),
+                        // The 'interval' here will make sure only one event is emitted every 100 ms
+                        Observable.interval(100, TimeUnit.MILLISECONDS)
+                )
+                .map {
+                    val mapItems = it.first
+
+                    return@map mapItems.filter(this@MapFragment::alreadyHasMarker)
+                }.filter {
+                    it.isNotEmpty()
+                }.subscribeBy {
+                    it.toObservable()
+                            .observeOn(AndroidSchedulers.mainThread())
```

Comment:
Just call `observeOn` in the chain before `subscribeBy` then you can just call `bindtoMap` directly.